src/TeamContributionCalendar/TeamContributionCalendar.js

```diff
@@ -159,22 +171,30 @@ export default class TeamContributionCalendar {
       if (data.error) {
         console.error(data.errorMessage);
       } else {
-        this.processGitLabCalendar(data.parsedCalendar, {
-          start: user.from,
-          end: user.to
-        });
+        const timeframe = { start: user.from, end: user.to };
+        // This sorts the days out falling out of the specified timeframe and restores the array of arrays structure(i.e. [[date, contributions], ...]) to key-value pairs of an object.
+        const filteredCalendar = Object.entries(data.parsedCalendar)
```

Comment:
How about creating a function for this that explains instead of the comment? Same for the above. If it's too complex to simply name and understand let's extract it and write specs for it.

Reply:
What do you mean by "write specs for it"?

Reply:
specs == tests

Reply:
Obsolete.